sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientTestBase.java

```diff
@@ -857,156 +561,113 @@ static void validateAnalysedSentiment(TextSentiment expectedSentiment, TextSenti
-        DetectedLanguage detectedLanguage3 = new DetectedLanguage("(Unknown)", "(Unknown)", 0.0);
-        List<DetectedLanguage> detectedLanguageList1 = Collections.singletonList(detectedLanguage1);
-        List<DetectedLanguage> detectedLanguageList2 = Collections.singletonList(detectedLanguage2);
-        List<DetectedLanguage> detectedLanguageList3 = Collections.singletonList(detectedLanguage3);
-
-        TextDocumentStatistics textDocumentStatistics1 = new TextDocumentStatistics(26, 1);
-        TextDocumentStatistics textDocumentStatistics2 = new TextDocumentStatistics(39, 1);
-        TextDocumentStatistics textDocumentStatistics3 = new TextDocumentStatistics(6, 1);
+    private static <T extends DocumentResult> void validateDocumentResult(boolean showStatistics,
+        DocumentResultCollection<T> expectedResults, DocumentResultCollection<T> actualResults,
+        BiConsumer<T, T> additionalAssertions) {
 
-        DetectLanguageResult detectLanguageResult1 = new DetectLanguageResult("0", textDocumentStatistics1, null, detectedLanguage1, detectedLanguageList1);
-        DetectLanguageResult detectLanguageResult2 = new DetectLanguageResult("1", textDocumentStatistics2, null, detectedLanguage2, detectedLanguageList2);
-        DetectLanguageResult detectLanguageResult3 = new DetectLanguageResult("2", textDocumentStatistics3, null, detectedLanguage3, detectedLanguageList3);
+        final Map<String, T> expected = expectedResults.stream().collect(Collectors.toMap(r -> r.getId(), r -> r));
+        final Map<String, T> actual = actualResults.stream().collect(Collectors.toMap(r -> r.getId(), r -> r));
 
-        TextDocumentBatchStatistics textDocumentBatchStatistics = new TextDocumentBatchStatistics(3, 3, 0, 3);
-        List<DetectLanguageResult> detectLanguageResultList = Arrays.asList(detectLanguageResult1, detectLanguageResult2, detectLanguageResult3);
+        assertEquals(expected.size(), actual.size());
 
-        return new DocumentResultCollection<>(detectLanguageResultList, MODEL_VERSION, textDocumentBatchStatistics);
-    }
+        if (showStatistics) {
+            validateBatchStatistics(expectedResults.getStatistics(), actualResults.getStatistics());
+        }
 
-    static DocumentResultCollection<RecognizeEntitiesResult> getExpectedBatchNamedEntities() {
-        NamedEntity namedEntity1 = new NamedEntity("Seattle", "Location", null, 26, 7, 0.80624294281005859);
-        NamedEntity namedEntity2 = new NamedEntity("last week", "DateTime", "DateRange", 34, 9, 0.8);
-        NamedEntity namedEntity3 = new NamedEntity("Microsoft", "Organization", null, 10, 9, 0.99983596801757812);
+        expected.forEach((key, expectedValue) -> {
+            T actualValue = actual.get(key);
 
-        List<NamedEntity> namedEntityList1 = Arrays.asList(namedEntity1, namedEntity2);
-        List<NamedEntity> namedEntityList2 = Collections.singletonList(namedEntity3);
+            assertTrue(actual.containsKey(key));
```

Comment:
This isn't needed. If the key doesn't exist, then the value returned will be null.